pytorch_lightning/trainer/callback_hook.py

```diff
@@ -31,7 +32,164 @@ class TrainerCallbackHookMixin(ABC):
     callbacks: List[Callback] = []
     lightning_module: "pl.LightningModule"
 
+    # TODO: Delete this in v1.8 (deprecation #10979)
```

Comment:
Please add formal deprecation warnings along with tests so users know they'll be removed and so we remember these todos later

Reply:
I initially did (still have it in my git stash) but then I thought it might be confusing since these hooks aren't deprecated like https://github.com/PyTorchLightning/pytorch-lightning/blob/753861a01f99543241aa4357e607ad65936c172d/pytorch_lightning/trainer/callback_hook.py#L243-L251, and I thought it might be misleading. But I can certainly add them back

Reply:
And if users use `call_hook` they will get a deprecation warning already

Reply:
In the deprecation message, you can say `"`TrainerCallbackHookMixin.<hook name>` was deprecated in v1.6 and will be removed in v1.8"` since we want to communicate that the class methods are being removed.

Reply:
Done. I put "removed in v1.7" since it is a mixin that really shouldn't be used by users right now, so not a typical deprecation.